src/HDF5.jl

```diff
@@ -2023,36 +2011,34 @@ function h5s_get_simple_extent_dims(space_id::Hid)
     h5s_get_simple_extent_dims(space_id, dims, maxdims)
     return tuple(reverse!(dims)...), tuple(reverse!(maxdims)...)
 end
+# Note: The following two functions implement direct ccalls because the binding generator
+# cannot (yet) do the string wrapping and memory freeing.
 function h5t_get_member_name(type_id::Hid, index::Integer)
-    pn = ccall((:H5Tget_member_name, libhdf5), Ptr{UInt8}, (Hid, Cuint), type_id, index)
+    pn = ccall((:H5Tget_member_name, libhdf5), Ptr{Cchar}, (Hid, Cuint), type_id, index)
     if pn == C_NULL
         error("Error getting name of compound datatype member #", index)
     end
     s = unsafe_string(pn)
-    Libc.free(pn)
-    s
+    h5_free_memory(pn)
+    return s
 end
 function h5t_get_tag(type_id::Hid)
     pc = ccall((:H5Tget_tag, libhdf5),
-                   Ptr{UInt8},
+                   Ptr{Cchar},
                    (Hid,),
                    type_id)
     if pc == C_NULL
         error("Error getting opaque tag")
     end
     s = unsafe_string(pc)
-    Libc.free(pc)
-    s
+    h5_free_memory(pc)
+    return s
 end
 
 function h5f_get_obj_ids(file_id::Hid, types::Integer)
-    sz = ccall((:H5Fget_obj_count, libhdf5), Int, (Hid, UInt32),
-               file_id, types)
-    sz >= 0 || error("error getting object count")
-    hids = Vector{Hid}(undef,sz)
-    sz2 = ccall((:H5Fget_obj_ids, libhdf5), Int, (Hid, UInt32, UInt, Ptr{Hid}),
-          file_id, types, sz, hids)
-    sz2 >= 0 || error("error getting objects")
+    sz = h5f_get_obj_count(file_id, type)
```

Comment:
```suggestion
    sz = h5f_get_obj_count(file_id, types)
```

Reply:
Note to self: this must not be tested for this typo to have been made.